doc/dev/error-categorization.md

```diff
@@ -0,0 +1,111 @@
+# Error Categorization
+
+In this document we start to explain how to categorize errors.
+Many errors are unique and can be difficult to categorize so this
+document will act as a guideline.
+
+## Categorization Mindset
+
+Errors along with their unique code only exist because they can be **reacted
+differently on programmatically**. Imagine a method call which returns a `Timeout` error. Naturally
+a senseful reaction would be to retry at a later point in time. So
+the reaction here would be to retry with a time based approach. On the other hand
+it does not make sense to differentiate between `No Write Permission` and `No Read Permission` as
+the application just knows that it simply cannot access the desired resource and tells
+the user to grant it. One could argue that an application could always react differently
+based on the error (eg. give yourself write or read permission) but such detailed granularity
+is rarely needed and would come with a lot of maintenance effort. In the previous versions of Elektra
+we were facing many duplicated errors because too many existed and developers simply cannot know each
+and every code.
+
+So if you feel the need for a new category you should first ask yourself the following question:
```

Comment:
If I already feel the need for a new category, I looked at the descriptions below  and decided my error doesn't fit any of these categories. Your question will not help me decide into which of the categories I should put my error. So my only way out is either creating a new category nonetheless, or somewhat randomly choose a category that fits a bit.

Reply:
What is your suggested way of finding an existing category or create a new one?